tests/unit/aws/protocol/test_parser.py

```diff
@@ -878,9 +878,10 @@ def test_restxml_headers_parsing():
     _botocore_parser_integration_test(
         service="s3",
         action="PutObject",
+        ContentLength=0,
```

Comment:
this addition is actually necessary, since now we set the content-length header correctly in our `Request` class, and the `content-length` header is translated into a member field `ContentLength` by the parser: https://github.com/boto/botocore/blob/b7a6f50cca3dcfdab7022a8e1ebe9e31edf57581/botocore/data/s3/2006-03-01/service-2.json#L8641-L8645